lib/game.rb

```diff
@@ -0,0 +1,51 @@
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS =[
+    [0,1,2],[3,4,5],[6,7,8], #HORIZONTAL WINNERS
+    [0,3,6],[1,4,7],[2,5,8], #VERTICAL WINNERS
+    [0,4,8],[2,4,6] #DIAGONAL WINNERS
+  ]
+
+  #///CLASS METHODS///#
+  def initialize(player_1=Player::Human.new("X"),player_2=Player::Human.new("O"),board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  #///INSTANCE METHODS ///#
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+  def over?
+    draw? || won?
+  end
+  def won?
+    #if any win combination has the same player token for all positions
+    #the game is won
+    WIN_COMBINATIONS.find do |win|
+      win.all? {|i| board.cells[i] == player_1.token} || win.all? {|i| board.cells[i] == player_2.token}
+    end
+  end
+  def draw?
+    @board.full? && !won?
+  end
+  def winner
+    @board.cells[won?[0]] if won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.